cmd/kube-proxy/app/server_others.go

```diff
@@ -91,8 +91,28 @@ func newProxyServer(
 		return nil, fmt.Errorf("unable to register configz: %s", err)
 	}
 
+	// detect IP Family
+	hostname, err := utilnode.GetHostname(config.HostnameOverride)
+	if err != nil {
+		return nil, err
+	}
+
+	client, eventClient, err := createClients(config.ClientConnection, master)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeIP := net.ParseIP(config.BindAddress)
+	if nodeIP.IsUnspecified() {
+		nodeIP = utilnode.GetNodeIP(client, hostname)
```

Comment:
Sigh. I was going to say `GetNodeIP` doesn't take a hostname, it takes a node name, but of course the docs are wrong and claim it takes a hostname. And you're just moving this code from elsewhere in the file so it doesn't make anything _more_ wrong.

Reply:
the terminology is confusing indeed, but I added a comment to try to clarify it